src/main/java/life/qbic/model/download/QbicDataDownloader.java

```diff
@@ -227,6 +227,35 @@ private void downloadFileFromDataStoreServer(DataSetFile dataSetFile, Path prefi
         final Path finalPath = OutputPathFinder.determineOutputDirectory(outputPath, prefix,
             file.getDataSetFile(), conservePaths);
         File newFile = new File(finalPath.toString());
+
+        if (newFile.exists()) {
+          try (
+              CheckedInputStream existingFileReader = new CheckedInputStream(
+                  Files.newInputStream(newFile.toPath()), new CRC32())) {
+
+            int bufferSize =
+                (newFile.length() < defaultBufferSize)
+                    ? (int) file.getDataSetFile().getFileLength()
+                    : defaultBufferSize;
+            byte[] buffer = new byte[bufferSize];
+            while (existingFileReader.read(buffer) != -1) {
+              // reading
+            }
+
+            ChecksumValidationResult checksumValidationResult = validateChecksum(
+                existingFileReader.getChecksum().getValue(), dataSetFile);
+            if (checksumValidationResult.isValid()) {
+              LOG.info("Found existing file with the identical content. Skipping "
+                  + finalPath.toAbsolutePath());
+              return;
+            } else {
+              LOG.info("Updating existing file " + finalPath.toAbsolutePath());
```

Comment:
I am thinking if this can be part of an application option, like `--update`? Otherwise this breaks the behavior of the app within a minor release, which should not happen.

Reply:
I don't see the break in behaviour. Currently all files are overwritten (updated) every time. This only skips the files where the result would be the previous state.
So the result of running the download command is the same.

Reply:
agreed